pkg/morserino/com_listen.go

```diff
@@ -25,33 +25,41 @@ THE SOFTWARE.
 import (
 	"fmt"
 	"io"
-	"log"
 	"strings"
 	"testing/iotest"
 
-	"github.com/on4kjm/morserino_display/pkg/morserino_console"
+	// "github.com/rs/zerolog"
 	"go.bug.st/serial"
 )
 
+const exitString string = "\nExiting...\n"
+
 // Main listen function with display to the console
-func ConsoleListen(morserinoPortName string, genericEnumPorts comPortEnumerator) error {
+func OpenAndListen(morserinoPortName string, genericEnumPorts comPortEnumerator, channels *MorserinoChannels) {
```

Comment:
I'm not in fond of this change: You're removing an `error` return and  instead  push down a set of channels that makes the code more complicated to understand.

How about keeping your error return and run the `OpenAndListen` in  an anonymous function like this:

```golang
var errCh = make(chan error)

// ...

go func() {
         if err := OpenAndListen(foo, bar); err != nil {
              errCh <- err
         }
}() 

```

Reply:
Also, most of your operations here are `initialization` phase error (aka misconfigured application for example) , not properly runtime. Those operations should be handled synchronously in your main goroutine I believe.

Maybe leverage a struct here, WDYT ? 

```golang
type Listenser struct {

}

// Initialization function for the initialization phase 
func NewListener(cfg Config) (*Listener, error) {
     // Initialize things that could fail synchronously
     smth, err := initializeSomething(cfg);
     if err != nil {
         return nil, err
     }
     
     return &Listener{smth: smth}, nil 
}
```

Another "smell" that makes me think this is your function name `OpenAndListen`:  you're doing two things in the same function that could be separated. 

Does it make sense ?